master/buildbot/test/unit/test_process_buildstep.py

```diff
@@ -120,6 +120,46 @@ def test_unexpectedKeywordArgument(self):
         self.assertRaisesConfigError("__init__ got unexpected keyword argument(s) ['oogaBooga']",
                 lambda: buildstep.BuildStep(oogaBooga=5))
 
+    def test_UrlKeywordReplacement(self):
+        """
+        Makes sure that URLs are being added properly and that keywords in a URL are replaced with actual values
+        """
+        class FakeFinishableStatus():
+            urls = {}
+            def addURL(self, name, url, results=None):
+                self.urls[name] = url
+
+        class FakeBuild():
```

Comment:
So is making fake classes like this the best way to test functions that use values from other classes? In this case, the URLs replace the keywords with values like build.builder.name and build.build_status.number, and I couldn't think of a better way to emulate those values than with fake classes.